File: CodeForces/155a/main.cpp
#include <iostream>
#include <stack>

int main() {

  int n, ans = 0;
  std::cin >> n;
  std::stack<int> max_till_now, min_till_now;

  for (int i = 0; i < n; i++) {
    int a;
    std::cin >> a;
    if (i == 0) {
      max_till_now.push(a);
      min_till_now.push(a);
    }
    if (a > max_till_now.top()) {
      ans++;
      max_till_now.push(a);
    } else if (a < min_till_now.top()) {
      ans++;
      min_till_now.push(a);
    }
  }

  std::cout << ans << std::endl;

  return 0;
}

File: CodeForces/155a/main.py
def inp():
    return (int(input()))


def inlt():
    return (list(map(int, input().split())))


def insr():
    return (input().strip())


n = inp()
arr = inlt()
max_till_now = [arr[0]]
min_till_now = [arr[0]]
ans = 0

for i in range(1, n):
    if (arr[i] > max_till_now[-1]):
        ans += 1
        max_till_now.append(arr[i])
    elif (arr[i] < min_till_now[-1]):
        ans += 1
        min_till_now.append(arr[i])

print(ans)
